### PR TITLE
feat(operator): adapt grad norm profiling for new partition strategy

### PR DESCRIPTION
**Pull request by JiaoPL**
Thanks for your contribution and we appreciate it a lot. The following instructions would make your pull request more healthy and more easily get feedback. If you do not understand some items, don't worry, just make the pull request and seek help from maintainers.

## Motivation

Please describe the motivation of this PR and the goal you want to achieve through this PR.

## Modification

Please briefly describe what modification is made in this PR.

## BC-breaking (Optional)

Does the modification introduce changes that break the backward compatibility of the downstream repositories?
If so, please describe how it breaks the compatibility and how the downstream projects should modify their code to keep compatibility with this PR.

## Use cases (Optional)

If this PR introduces a new feature, it is better to list some use cases here and update the documentation.

## Checklist

**Before PR**:

- [ ] Pre-commit or other linting tools are used to fix the potential lint issues.
- [ ] Bug fixes are fully covered by unit tests, the case that causes the bug should be added in the unit tests.
- [ ] The modification is covered by complete unit tests. If not, please add more unit test to ensure the correctness.
- [ ] The documentation has been modified accordingly, like docstring or example tutorials.

**After PR**:

- [ ] If the modification has potential influence on downstream or other related projects, this PR should be tested with those projects.
- [ ] CLA has been signed and all committers have signed the CLA in this PR.

### CODE REVIEW

File: internlm/solver/optimizer/hybrid_zero_optim.py [hunk 1]
```diff
@@ -661,13 +661,26 @@ def _compute_param_norm_stage(
...
             grads = [self.padding_grad.to(dtype)]
             params = [self.padding_tensor.to(dtype)]
 
+            if group_id == 0:
+                for param in params:
+                    if self.use_isp:
+                        setattr(param, IS_WEIGHT_ZERO_PARALLEL, True)
+                    else:
+                        setattr(param, IS_TENSOR_ZERO_PARALLEL, True)
+            elif group_id == 1:
```
Comment (huangting4201) [inline internlm/solver/optimizer/hybrid_zero_optim.py]: 等moe的pr合入后，这里需要更新

File: internlm/solver/optimizer/hybrid_zero_optim.py [hunk 2]
```diff
@@ -676,17 +689,43 @@ def _compute_param_norm_stage(
...
```
Comment (huangting4201) [inline internlm/solver/optimizer/hybrid_zero_optim.py]: 同上

File: internlm/solver/optimizer/utils.py [hunk 3]
```diff
@@ -247,7 +254,7 @@ def append_grad(g, p):
         elif only_output:
             param_name = p.param_name if hasattr(p, "param_name") else "unknown-padding"
             if (
-                gpc.config.model["vocab_size"] == g.shape[0] * gpc.get_world_size(ParallelMode.TENSOR)
+                gpc.config.model["vocab_size"] == g.shape[0] * gpc.get_world_size(param_parallel_mode)
```
Comment (huangting4201) [inline internlm/solver/optimizer/utils.py]: 我们的head层无论是isp算法还是非isp，走的都是tp并行，所以这里我理解应该仍然是ParallelMode.TENSOR，不用变动？

File: internlm/solver/optimizer/hybrid_zero_optim.py (hunk 1 above)
Comment (JiaoPL) [reply internlm/solver/optimizer/hybrid_zero_optim.py]: 已更新

File: internlm/solver/optimizer/hybrid_zero_optim.py (hunk 2 above)
Comment (JiaoPL) [reply internlm/solver/optimizer/hybrid_zero_optim.py]: 已更新

File: internlm/solver/optimizer/utils.py (hunk 3 above)
Comment (JiaoPL) [reply internlm/solver/optimizer/utils.py]: 改回来了

**Pull request merged**